napari/layers/shapes/_shapes_models/shape.py

```diff
@@ -280,31 +339,107 @@ def _set_meshes_compiled(
+        closed : bool
+            Bool which determines if the edge is closed or not
+        face : bool
+            Bool which determines if the face need to be traingulated
+        edge : bool
+            Bool which determines if the edge need to be traingulated
+        """
+        if data.shape[1] == 3:
+            self._set_meshes_py(data, closed=closed, face=face, edge=edge)
+            return
+
+        # if we are computing both edge and face triangles, we can do so
+        # with a single call to the compiled backend
+        if edge and face:
+            try:
+                (triangles, vertices), (centers, offsets, edge_triangles) = (
+                    partsegcore_triangulate.triangulate_polygon_with_edge_numpy_li(
+                        [data]
+                    )
+                )
+            except BaseException as e:  # pragma: no cover
+                path, text_path = _save_failed_triangulation(
+                    data, backend='partsegcore'
+                )
+                raise RuntimeError(
+                    f'Triangulation failed. Data saved to {path} and {text_path}'
+                ) from e
+
+            self._edge_vertices = centers
+            self._edge_offsets = offsets
+            self._edge_triangles = edge_triangles
             self._face_vertices = vertices
             self._face_triangles = triangles
+            return
+
+        # otherwise, we make individual calls to specialized functions
+        if edge:
+            self._edge_vertices, self._edge_offsets, self._edge_triangles = (
+                partsegcore_triangulate.triangulate_path_edge_numpy(
+                    data, closed=closed
+                )
+            )
+        else:
+            self._set_empty_edge()
+        if face:
+            self._face_triangles, self._face_vertices = (
+                partsegcore_triangulate.triangulate_polygon_numpy_li([data])
+            )
         else:
-            self._face_vertices = np.empty((0, self.ndisplay))
-            self._face_triangles = np.empty((0, 3), dtype=np.uint32)
+            self._set_empty_face()
 
-    def _set_meshes(  # noqa: B027
+    def _set_meshes_triangle(
         self,
         data: npt.NDArray,
         closed: bool = True,
         face: bool = True,
         edge: bool = True,
-    ) -> None: ...
+    ) -> None:
+        """Sets the face and edge meshes from a set of points.
+
+        Uses the triangle package to triangulate the polygon face
+
+        Parameters
+        ----------
+        data : np.ndarray
+            Nx2 or Nx3 array specifying the shape to be triangulated
+        closed : bool
+            Bool which determines if the edge is closed or not
+        face : bool
+            Bool which determines if the face need to be traingulated
+        edge : bool
+            Bool which determines if the edge need to be traingulated
+        """
+        self._set_meshes_py(data, closed=closed, face=face, edge=edge)
```

Comment:
This is a dummy implementation until #6654 is merged to avoid creating merge conflicts.

Reply:
wait are we merging this before #6654 or vice versa?

Reply:
Oh I was confused because of this comment:

https://github.com/napari/napari/pull/6654#discussion_r2032788275

Why not have the right naming directly in #6654?